### DISCUSSION TITLE
LaTeX command to DocBook element

### DISCUSSION THREAD

**Discussion by raghnysh** [jgm/pandoc#10332] [Q&A] [answered]
Is it possible to transform a custom LaTeX command ```\firstterm``` to the DocBook element ```firstterm```, so that one gets 

```
$ echo "Hello, \firstterm{world}!" | pandoc -f latex -t docbook
<para>
  Hello, <firstterm>world</firstterm>!
</para>
```

instead of

```
$ echo "Hello, \firstterm{world}!" | pandoc -f latex -t docbook
<para>
  Hello, !
</para>
```

**Comment by jgm** [comment, marked as answer]
You'd have to (a) use `-f latex+raw_tex`, and (b) create a Lua filter that changes RawInline elements that match this pattern into the thing you desire.

**Reply by raghnysh** [reply]
Thanks, I'll try that.

**Comment by raghnysh** [comment]
```
$ cat filter.lua 
function RawInline(elem)
   return pandoc.RawInline('docbook', elem.text:gsub("\\(firstterm){(.-)}", "<%1>%2</%1>"))
end
```

```
$ echo "Hello \firstterm{world}!" | pandoc -f latex+raw_tex --lua-filter=filter.lua -t docbook
<para>
  Hello <firstterm>world</firstterm>!
</para>
```
Thanks for your help!

**Comment by bpj** [comment]
You would want to check that the raw inline is indeed (La)TeX:

``````
if 'latex' == elem.format or 'tex' == elem.format then
  ...
end
``````

It might not be necessary in the current case, but it is always better to be sure.

**Reply by raghnysh** [reply]
Yes, thanks for the tip.
